### PR TITLE
Fixing inertia calcs

### LINKED ISSUE #17: Inertia Tensor needs to be about center of mass
According to [this](http://sdformat.org/spec?elem=link), the inertia tensor for the a link in the sdf needs to be about the center of mass of the link. Since the current tensor is calculated about the wing root, the calculated tensor needs to be manipulated according to the [parallel axis theorem](https://en.wikipedia.org/wiki/Parallel_axis_theorem).

### PR DESCRIPTION
**Pull request by BrightKnight02**
Trying to fix the inertia calculations for the optimization. The was a small error in the slender rod calculations, which was found after creating the tests in `test_inertial_preperties.py`. 

I'm not going to close the issue yet as I may be interpreting some things wrong.

There is a chance things that were not supposed to change did change. For those things I am sorry.

### CODE REVIEW

File: flapping_proj/unit_tests/test_inertial_preperties.py
Comment (darint916) [inline flapping_proj/unit_tests/test_inertial_preperties.py:1]: Write failing asserts? for the unit test library to catch failures

File: flapping_proj/unit_tests/test_inertial_preperties.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                              [  0,   0],
+                              [  0,   1],
+                              [  1,   0]])
+        for i in range(len(self.test_coordinates)):
+            coords = self.test_coordinates[i]
+            calc_com = line_com(coords[0], coords[1], coords[2], coords[3], diameter=self.rod_diameter, density = self.rod_density)
+            calc_com = calc_com / self.truth_mass[i]
+            Message.info(f"Analysing rod with endpoints (y, z) at ({coords[0]}, {coords[1]}) and ({coords[2]}, {coords[3]})")
+            
+            if almost_equal(truth_com[i][0], calc_com[0], self.error_tol) and almost_equal(truth_com[i][1], calc_com[1], self.error_tol):
```
Comment (darint916) [inline flapping_proj/unit_tests/test_inertial_preperties.py]: for assert do:
self.assertTrue(almost_equal(truth_com[i][0], calc_com[0], self.error_tol) and almost_equal(truth_com[i][1], calc_com[1], self.error_tol, Message.error("Failed case"))

**Review by darint916 — CHANGES_REQUESTED** [review]
small changes in formatting and fixing testing

File: flapping_proj/unit_tests/test_inertial_preperties.py (hunk 1 above)
Comment (BrightKnight02) [reply flapping_proj/unit_tests/test_inertial_preperties.py]: Updated with https://github.com/darint916/ros2-gazebo-drake-sims/pull/19/commits/7cca65e214853907c57dae233d516940d168f046

**Comment by darint916** [comment]
Test cases passed unit tests, but failed on top-level smoke test, new test case was made with given opt parameters:
0.00484565 -0.00135816  0.03554422 -0.06507584  0.09359727 -0.00165281 0.51404022
produced values of 
'ixx': 3.392129987389574e-06, 'ixy': 0.0, 'ixz': 0.0, 'iyy': 1.9036163412645447e-06, 'iyz': -2.629275870389845e-06, 'izz': 1.4885136461250288e-06

Solidworks truth values as:
Moments of inertia: ( grams *  square millimeters )
Taken at the center of mass and aligned with the output coordinate system. (Using positive tensor notation.)
    Lxx = 197.457736    Lxy = 0.000000    Lxz = 0.000000
    Lyx = 0.000000    Lyy = 73.398949    Lyz = 23.131955
    Lzx = 0.000000    Lzy = 23.131955    Lzz = 124.076478

**Review by darint916 — APPROVED** [review]

**Review by darint916 — CHANGES_REQUESTED** [review]
Test cases passed unit tests, but failed on top-level smoke test, new test case was made with given opt parameters:
`0.00484565 -0.00135816  0.03554422 -0.06507584  0.09359727 -0.00165281 0.51404022`
produced values of 
`'ixx': 3.392129987389574e-06, 'ixy': 0.0, 'ixz': 0.0, 'iyy': 1.9036163412645447e-06, 'iyz': -2.629275870389845e-06, 'izz': 1.4885136461250288e-06`

Solidworks truth values as:
Moments of inertia: ( grams *  square millimeters )
Taken at the center of mass and aligned with the output coordinate system. (Using positive tensor notation.)
 `   Lxx = 197.457736    Lxy = 0.000000    Lxz = 0.000000
    Lyx = 0.000000    Lyy = 73.398949    Lyz = 23.131955
    Lzx = 0.000000    Lzy = 23.131955    Lzz = 124.076478`
    
    
    All of Inertia Properties of Rod , COM, total mass, moment of inertia tested.
    
    Seems to be failing on the film.

**Comment by BrightKnight02** [comment]
This is ready for an additional review. https://github.com/darint916/ros2-gazebo-drake-sims/pull/19/commits/d1ec297ed6a17289ca889a7ec48cf7fd150428fd should have fixed all calculations related to the current optimization.

**Comment by BrightKnight02** [comment]
The inertia was changed to be in the correct coordinate system. Inertia and center of mass was calculated as if the wing spar was along the y axis, but Gazebo was set up such that the wing spar was along the x axis. This was fixed by changing what index lines 83-88 and line 92 in `top_level.py` pointed to for the inertia tensor calculated by the `TriWing` class. This is equivalent to rotating the coordinate system.

**Review by darint916 — APPROVED** [review]

**Review by darint916 — APPROVED** [review]

**Pull request merged**